pkg/resource/deploy/step_generator.go

```diff
@@ -849,7 +849,7 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 	// We will also not record this non-created resource into the checkpoint as it doesn't actually
 	// exist.
 
-	if !isTargeted && !providers.IsProviderType(goal.Type) {
+	if !isTargeted {
```

Comment:
tl;dr

previously: non-targeted resources are skipped as long as they're not providers.
now: all non-targeted resources are skipped.